modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java

```diff
@@ -969,4 +977,44 @@ IndexMeta createIndexMetaForNewIndex(int indexId) {
             return sortedIndexes.get(indexId);
         });
     }
+
+    /**
+     * Destroys an index storage identified by the given index ID.
+     *
+     * @param indexId Index ID which storage will be destroyed.
+     * @return Future that will be completed as soon as the storage has been destroyed.
+     */
+    public CompletableFuture<Void> destroyIndex(int indexId) {
+        return busy(() -> {
+            throwExceptionIfStorageNotInRunnableState();
+
+            CompletableFuture<Void> result = nullCompletedFuture();
+
+            PageMemoryHashIndexStorage hashIndexStorage = hashIndexes.remove(indexId);
+
+            if (hashIndexStorage != null) {
+                assert !sortedIndexes.containsKey(indexId);
```

Comment:
Are you sure a check is needed?

Reply:
Yes, it's just a consistency assertion, wouldn't hurt

Reply:
Can such a check be added when adding indexes? 
It’s strange that it’s only checked here.

Reply:
Why is it strange, it's just an assertion to be sure that we don't have some kind of a bug